spec.bs

```diff
@@ -1145,38 +1148,45 @@ steps. They return an [=aggregation coordinator=], null or a {{DOMException}}:
 1. Return |origin|.
 
 To <dfn algorithm>obtain the pre-specified report parameters</dfn> given a
-{{SharedStorageRunOperationMethodOptions}} |options|, perform the following
-steps. They return a [=pre-specified report parameters=], null, or a
+{{SharedStorageRunOperationMethodOptions}} |options| and a [=Window/browsing context=] |context|,
+perform the following steps. They return a [=pre-specified report parameters=], null, or a
 {{DOMException}}:
 1. If |options|["{{SharedStorageRunOperationMethodOptions/privateAggregationConfig}}"]
     does not [=map/exist=], return null.
 1. Let |privateAggregationConfig| be
     |options|["{{SharedStorageRunOperationMethodOptions/privateAggregationConfig}}"].
+1. Let |calledFromFencedFrame| be true if |context|'s [=browsing context/fenced frame config
+    instance=] is not null, false otherwise.
 1. Let |contextId| be null.
 1. If |privateAggregationConfig|["{{SharedStoragePrivateAggregationConfig/contextId}}"]
     [=map/exists=], set |contextId| to
     |privateAggregationConfig|["{{SharedStoragePrivateAggregationConfig/contextId}}"].
 1. If |contextId|'s [=string/length=] is greater than 64, return a new
     {{DOMException}} with name "`DataError`".
+1. If |calledFromFencedFrame| is true and |contextId| is not null, return a new {{DOMException}}
+    with name "`DataError`".
 1. Let |filteringIdMaxBytes| be the [=default filtering ID max bytes=].
 1. If |privateAggregationConfig|["{{SharedStoragePrivateAggregationConfig/filteringIdMaxBytes}}"]
     [=map/exists=], set |filteringIdMaxBytes| to
     |privateAggregationConfig|["{{SharedStoragePrivateAggregationConfig/filteringIdMaxBytes}}"].
 1. If |filteringIdMaxBytes| is not [=set/contained=] in the [=valid filtering ID
     max bytes range=], return a new {{DOMException}} with name "`DataError`".
+1. If |calledFromFencedFrame| is true and |filteringIdMaxBytes| is not the [=default filtering ID
```

Comment:
nit: Maybe we could consolidate these into something like:
```
1. If |context|'s ... instance=] is not null:
  1. If |contextId| is not null or ..., return a new {{DOMException}} ...
```

Might slightly help readability if we put them together, wdyt? We could also consider adding a note explaining the why.

Reply:
Consolidated into one section in a nested if statement.